src/pybind/mgr/influx/module.py

```diff
@@ -387,7 +387,7 @@ def shutdown(self) -> None:
         self.log.debug('Shutting down queue workers')
 
         for _ in self.workers:
-            self.queue.put(None)
+            self.queue.put([])
```

Comment:
This changes the logic. `None` was used as a special value to indicate worker shut down. The previous value `[]` has now become a special case. 

I'd :+1: to keep the original behavior.

Reply:
@sebastian-philipp you are right. there is chance that we have no pools in cluster in that case, we should not signal the workers to stop and exit. unless we never push an empty list of data points to the queue if there is no data to push. but i don't want to change the existing behavior which writes empty data points to influx db even if there is no data.

so i changed the type of `self.queue` to `queue.Queue[Optional[List[Dict[str, str]]]]`. not a good example of conciseness, but at least it's correct =)